src/gui/entry/EntryView.cpp

```diff
@@ -144,24 +268,262 @@ Entry* EntryView::entryFromIndex(const QModelIndex& index)
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Toggle visibility of column referenced by triggering action
+ */
+void EntryView::toggleColumnVisibility(QAction *action)
+{
+    /*
+     * Verify action carries a column index as data. Since QVariant.toInt()
+     * below will accept anything that's interpretable as int, perform a type
+     * check here to make sure data actually IS int
+     */
+    if (static_cast<QMetaType::Type>(action->data().type()) != QMetaType::Int) {
+        Q_ASSERT(false);
+        return;
+    }
+
+    /*
+     * Toggle column visibility. Visible columns will only be hidden if at
+     * least one visible column remains, as the table header will disappear
+     * entirely when all columns are hidden, rendering the context menu in-
+     * accessible
+     */
+    int colidx = action->data().toInt();
+    if (action->isChecked()) {
+        header()->showSection(colidx);
+        if (header()->sectionSize(colidx) == 0) {
+            header()->resizeSection(colidx, header()->defaultSectionSize());
+        }
+    }
+    else {
+        if ((header()->count() - header()->hiddenSectionCount()) > 1) {
+            header()->hideSection(colidx);
+        }
+        else {
+            action->setChecked(true);
+        }
+    }
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Resize columns to fit all visible columns within the available space
+ *
+ * NOTE:
+ * If EntryView::resizeEvent() is overridden at some point in the future,
+ * its implementation MUST call the corresponding parent method using
+ * 'QTreeView::resizeEvent(event)'. Without this, fitting to window will
+ * be broken and/or work unreliably (stumbled upon during testing)
+ *
+ * NOTE:
+ * Testing showed that it is absolutely necessary to emit signal 'viewState
+ * Changed' here. Without this, an incomplete view state might get saved by
+ * 'DatabaseWidgetStateSync' (e.g. only some columns resized)
+ */
+void EntryView::fitColumnsToWindow()
+{
+    header()->resizeSections(QHeaderView::Stretch);
+    emit viewStateChanged();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Resize columns to fit current table contents, i.e. make all contents
+ * entirely visible
+ */
+void EntryView::fitColumnsToContents()
+{
+    /* Resize columns to fit contents */
+    header()->resizeSections(QHeaderView::ResizeToContents);
+
+    /*
+     * Determine total width of currently visible columns. If there is
+     * still some space available on the header, equally distribute it to
+     * visible columns and add remaining fraction to last visible column
+     */
+    int width = 0;
+    for (int colidx = 0; colidx < header()->count(); colidx++) {
+        if (!header()->isSectionHidden(colidx)) {
+            width += header()->sectionSize(colidx);
+        }
+    }
+    int visible = header()->count() - header()->hiddenSectionCount();
+    int avail = header()->width() - width;
+    if ((visible > 0) && (avail > 0)) {
```

Comment:
I think you can invert the condition and safely return here.

Reply:
Agreed.